### PR TITLE
N21-1250 CTL board element

### PR DESCRIPTION
**Pull request by MarvinOehlerkingCap**
# Short Description
- adds an external tool element to boards

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-1250
https://github.com/hpi-schul-cloud/dof_app_deploy/pull/644
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4419

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/data-board/BoardApi.composable.unit.ts [hunk 1]
```diff
@@ -184,6 +184,29 @@ describe("BoardApi.composable", () => {
 			);
 		});
 
+		it("should call elementControllerUpdateElement api with ExternalToolElement", async () => {
+			const { updateElementCall } = useBoardApi();
+			const PAYLOAD = {
```
Comment (arnegns) [inline src/components/data-board/BoardApi.composable.unit.ts:189]: why is the variable capitalized?

File: src/components/feature-board/shared/AddElementDialog.vue [hunk 2]
```diff
@@ -73,7 +73,9 @@ export default defineComponent({
 		const submissionsEnabled =
 			envConfigModule.getEnv.FEATURE_COLUMN_BOARD_SUBMISSIONS_ENABLED;
 
-		const dialogWidth = submissionsEnabled ? 426 : 320;
+		const dialogWidth: ComputedRef<number> = computed(() =>
```
Comment (arnegns) [inline src/components/feature-board/shared/AddElementDialog.vue:76]: Something for north* @OliverHappe @uidp

File: src/components/page-board/ColumnBoard.page.vue [hunk 3]
```diff
@@ -3,27 +3,53 @@
...
+		const contextExternalToolsModule: ContextExternalToolsModule = injectStrict(
+			CONTEXT_EXTERNAL_TOOLS_MODULE_KEY
+		);
+		const envConfigModule: EnvConfigModule = injectStrict(
+			ENV_CONFIG_MODULE_KEY
+		);
 
 		useTitle(pageTitle);
 
+		onMounted(async () => {
```
Comment (arnegns) [inline src/components/page-board/ColumnBoard.page.vue:42]: can we load this if we have the first element from type external tool?

**Review by arnegns — CHANGES_REQUESTED** [review]

File: src/components/data-board/BoardApi.composable.unit.ts (hunk 1 above)
Comment (MarvinOehlerkingCap) [reply src/components/data-board/BoardApi.composable.unit.ts:189]: idk why they did it like that, but it is the wrong style, so I will change it.

**Review by arnegns — APPROVED** [review]

**Review by arnegns — APPROVED** [review]

File: src/components/feature-board/shared/AddElementDialog.vue (hunk 2 above)
Comment (uidp) [reply src/components/feature-board/shared/AddElementDialog.vue:76]: Hm, what happens, when there are a lot more `elementTypeOptions`?

Comment (arnegns) [reply src/components/feature-board/shared/AddElementDialog.vue:76]: @uidp the number of elementTypeOptions is fixed at the moment or?

**Pull request merged**